frontend/src/components/bucket/BucketPermission.vue

```diff
@@ -0,0 +1,156 @@
+import { useBucketStore } from '@/store/bucketStore';
+import { Permissions } from '@/utils/constants';
+
+const props = defineProps<{
+  bucketId: string;
+}>();
+
+const bucketStore = useBucketStore();
+const { loading, permissions } = storeToRefs(useBucketStore());
+
+onMounted(() => {
+  bucketStore.getBucketPermissions(props.bucketId);
+});
+
+const updateBucketPermission = (value: any, userId: string, permCode: string) => {
+  if (value) {
+    bucketStore.addBucketPermission(props.bucketId, userId, permCode);
+  } else {
+    bucketStore.deleteBucketPermission(props.bucketId, userId, permCode);
+  }
+};
+
+const removeBucketUser = (userId: string) => {
+  bucketStore.removeBucketUser(props.bucketId, userId);
+};
+</script>
+
+<template>
+  <Button class="mt-1 mb-4"> <font-awesome-icon icon="fa-solid fa-user-plus" class="mr-1" /> Add user </Button>
+
+  <DataTable
+    :loading="loading"
+    :value="permissions"
+    dataKey="bucketId"
+    class="p-datatable-sm"
+    stripedRows
+    responsiveLayout="scroll"
+    :paginator="true"
+    :rows="10"
+    paginatorTemplate="RowsPerPageDropdown CurrentPageReport PrevPageLink NextPageLink "
+    currentPageReportTemplate="{first}-{last} of {totalRecords}"
+    :rowsPerPageOptions="[10, 20, 50]"
+  >
+    <template #empty>
+      <div class="flex justify-content-center">
+        <h3>There are no users associated with this bucket.</h3>
+      </div>
+    </template>
+    <Column field="fullName" header="Name" />
+    <Column header="Upload" bodyClass="content-center">
+      <template #body="{ data }">
+        <Checkbox
+          inputId="create"
+          v-model="data.create"
+          :binary="true"
+          @input="(value) => updateBucketPermission(value, data.userId, Permissions.CREATE)"
+        />
+      </template>
+    </Column>
+    <Column header="Read" bodyClass="content-center">
+      <template #body="{ data }">
+        <Checkbox
+          inputId="read"
+          v-model="data.read"
+          :binary="true"
+          @input="(value) => updateBucketPermission(value, data.userId, Permissions.READ)"
+        />
+      </template>
+    </Column>
+    <Column header="Update" bodyClass="content-center">
+      <template #body="{ data }">
+        <Checkbox
+          inputId="update"
+          v-model="data.update"
+          :binary="true"
+          @input="(value) => updateBucketPermission(value, data.userId, Permissions.UPDATE)"
+        />
+      </template>
+    </Column>
+    <Column header="Delete" bodyClass="content-center">
+      <template #body="{ data }">
+        <Checkbox
+          inputId="delete"
+          v-model="data.delete"
+          :binary="true"
+          @input="(value) => updateBucketPermission(value, data.userId, Permissions.DELETE)"
+        />
+      </template>
+    </Column>
+    <Column header="Manage" bodyClass="content-center">
+      <template #body="{ data }">
+        <Checkbox
+          inputId="manage"
```

Comment:
not sure if you planned to disable current user from removing themself
maybe you can do something like `disabled=data.userId === _keycloak.userId`
and you could disable the 'remove user' button

Reply:
Future consideration. I've had discussion with Tyler and Sharolyn regarding this, and we are not blocking any functionality at this time.